pkg/providers/launchtemplate/suite_test.go

```diff
@@ -1383,8 +1405,201 @@ var _ = Describe("LaunchTemplates", func() {
+							v1.ResourceEphemeralStorage: resource.MustParse("2Gi"),
+						},
+					}),
+					Entry("evictionHard", "evictionHard", corev1beta1.KubeletConfiguration{
+						EvictionHard: map[string]string{
+							"memory.available":  "10%",
+							"nodefs.available":  "15%",
+							"nodefs.inodesFree": "5%",
+						},
+					}),
+					Entry("evictionSoft", "evictionSoft", corev1beta1.KubeletConfiguration{
+						EvictionSoft: map[string]string{
+							"memory.available":  "10%",
+							"nodefs.available":  "15%",
+							"nodefs.inodesFree": "5%",
+						},
+						EvictionSoftGracePeriod: map[string]metav1.Duration{
+							"memory.available":  {Duration: time.Minute},
+							"nodefs.available":  {Duration: time.Second * 180},
+							"nodefs.inodesFree": {Duration: time.Minute * 5},
+						},
+					}),
+					Entry("evictionSoftGracePeriod", "evictionSoftGracePeriod", corev1beta1.KubeletConfiguration{
+						EvictionSoft: map[string]string{
+							"memory.available":  "10%",
+							"nodefs.available":  "15%",
+							"nodefs.inodesFree": "5%",
+						},
+						EvictionSoftGracePeriod: map[string]metav1.Duration{
+							"memory.available":  {Duration: time.Minute},
+							"nodefs.available":  {Duration: time.Second * 180},
+							"nodefs.inodesFree": {Duration: time.Minute * 5},
+						},
+					}),
+					Entry("evictionMaxPodGracePeriod", "evictionMaxPodGracePeriod", corev1beta1.KubeletConfiguration{
+						EvictionMaxPodGracePeriod: lo.ToPtr[int32](300),
+					}),
+					Entry("podsPerCore", "podsPerCore", corev1beta1.KubeletConfiguration{
+						PodsPerCore: lo.ToPtr[int32](2),
+					}),
+					Entry("clusterDNS", "clusterDNS", corev1beta1.KubeletConfiguration{
+						ClusterDNS: []string{"10.0.100.0"},
+					}),
+					Entry("imageGCHighThresholdPercent", "imageGCHighThresholdPercent", corev1beta1.KubeletConfiguration{
+						ImageGCHighThresholdPercent: lo.ToPtr[int32](50),
+					}),
+					Entry("imageGCLowThresholdPercent", "imageGCLowThresholdPercent", corev1beta1.KubeletConfiguration{
+						ImageGCLowThresholdPercent: lo.ToPtr[int32](50),
+					}),
+					Entry("cpuCFSQuota", "cpuCFSQuota", corev1beta1.KubeletConfiguration{
+						CPUCFSQuota: lo.ToPtr(false),
+					}),
+				)
+			})
+			It("should set LocalDiskStrategy to Raid0 when specified by the InstanceStorePolicy", func() {
+				nodeClass.Spec.InstanceStorePolicy = lo.ToPtr(v1beta1.InstanceStorePolicyRAID0)
+				ExpectApplied(ctx, env.Client, nodeClass, nodePool)
+				pod := coretest.UnschedulablePod()
+				ExpectProvisioned(ctx, env.Client, cluster, cloudProvider, prov, pod)
+				ExpectScheduled(ctx, env.Client, pod)
+				for _, userData := range ExpectLaunchTemplatesCreatedWithUserData() {
+					configs := ExpectUserDataCreatedWithNodeConfigs(userData)
+					Expect(len(configs)).To(Equal(1))
+					Expect(configs[0].Spec.Instance.LocalStorage.Strategy).To(Equal(admv1alpha1.LocalStorageRAID0))
+				}
 			})
+			DescribeTable(
+				"should merge custom user data",
```

Comment:
Just checking here: we don't need to check if the kubelet config will be merged properly because that'll be done by nodeadm, not us, right?

Reply:
Yep, we just need to check that both the Karpenter config and the user config are correctly added to the MIME archive, nodeadm handles merging the two.